host/class/uvc/usb_host_uvc_2/include/usb/uvc_host.h

```diff
@@ -0,0 +1,241 @@
+                                          `uvc_host_stream_config_t.advanced` to allocate additional buffers. */
+};
+
+/**
+ * @brief UVC Device Event data structure
+ */
+typedef struct {
+    enum uvc_host_dev_event type;
+    union {
+        int error;                         //!< Error code from USB Host
+        uvc_host_stream_hdl_t stream_hdl;  //!< Disconnection event
+    } data;
+} uvc_host_stream_event_data_t;
+
+/**
+ * @brief Formats supported by this driver
+ */
+enum uvc_host_stream_format {
+    UVC_VS_FORMAT_UNDEFINED = 0, // Invalid format. Do not request this format from the camera.
+    UVC_VS_FORMAT_MJPEG,
+    UVC_VS_FORMAT_YUY2,
+    UVC_VS_FORMAT_H264,
+    UVC_VS_FORMAT_H265,
+};
+
+typedef struct {
+    unsigned h_res;                     /**< Horizontal resolution */
+    unsigned v_res;                     /**< Vertical resolution */
+    float fps;                          /**< Frames per second */
+    enum uvc_host_stream_format format; /**< Frame coding format */
+} uvc_host_stream_format_t;
+
+/**
+ * @brief Video Stream frame
+ *
+ * This type is returned from frame callback upon receiving new frame
+ */
+typedef struct {
+    const uvc_host_stream_format_t vs_format; /**< Format of this frame buffer */
+    size_t data_buffer_len;                   /**< Max data length supported by this frame buffer */
+    size_t data_len;                          /**< Data length of currently store frame */
+    uint8_t *data;                            /**< Frame data */
+} uvc_host_frame_t;
+
+/**
+ * @brief Stream event callback type
+ *
+ * @param[in] event    Event structure
+ * @param[in] user_ctx User's argument passed to open function
+ */
+typedef void (*uvc_host_stream_callback_t)(const uvc_host_stream_event_data_t *event, void *user_ctx);
+
+/**
+ * @brief Frame callback type
+ *
+ * @param[in] frame    Received frame
+ * @param[in] user_ctx User's argument passed to open function
+ * @return
+ *     - true:  The frame was processed by user. The UVC driver owns the frame now.
+ *     - false: The frame was not yet processed by user.
+ *              The user must call uvc_host_frame_return() to return it to UVC driver.
+ */
+typedef bool (*uvc_host_frame_callback_t)(const uvc_host_frame_t *frame, void *user_ctx);
+
+/**
+ * @brief Configuration structure of UVC device
+ *
+ */
+typedef struct {
+    uvc_host_stream_callback_t event_cb;  /**< Stream's event callback function. Can be NULL */
+    uvc_host_frame_callback_t frame_cb;   /**< Stream's frame callback function */
+    void *user_ctx;                       /**< User's argument that will be passed to the callbacks */
+    struct {
+        uint16_t vid;                     /**< Device's Vendor ID. Set to 0 for any */
+        uint16_t pid;                     /**< Device's Product ID. Set to 0 for any */
+        uint8_t uvc_function_index;       /**< Index of UVC function you want to use. Set to 0 to use first available UVC function */
```

Comment:
Is that `bInterfaceNum` ?

Reply:
No, it isn't. The camera function usually have more (typically 2) interfaces binded together with IAD, so it would be difficult to identify the stream with bInterfaceNum.

Next, the USB cameras often implement microphone with audio interfaces.

So, if you wanted to open a Video stream you'd have to know which bInterfaceNum you need to open. This is too inflexbile for users who just want to open the first available Video stream.

That is what `uvc_function_index` does. 0 = open first available Video stream. 1 =  open second available Video stream...

Reply:
Got it. Then maybe? 

```suggestion
        uint8_t uvc_stream_index;       /**< Index of UVC stream you want to use. Set to 0 to use first available UVC stream */
```